scripts/functional-tests.sh

```diff
@@ -8,6 +15,11 @@ export SCRIPTDIR=$(dirname "$0")
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
+# Support overriding the test directory for local testing otherwise creates a temporary directory
+TEST_DIR=${TEST_DIR:-$(mktemp -d --tmpdir=/var/tmp instructlab-test-tmp.XXXXXX)}
```

Comment:
You should not assume `/var/tmp`, or `/tmp`, exists. (POSIX [leaves](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/xrat/V4_xbd_chap10.html) `/tmp` in the standard for historical reasons, but it's not advised to use it anyway.)

`mktemp` will already use `TMPDIR` to decide where to create a temporary file. We should allow the user to specify a different `TMPDIR` to use. They can already do it by `TMPDIR=/var/tmp ./scripts/functional-tests.sh` (but they cannot do it when they run the tests with `tox -e py311-functional` because `tox` strips off all envvars except those listed in `passenv` - hence my suggestion to allow the variable to be passed into the functional `tox` env.)

Reply:
And `TEST_DIR` and `SKIP_TEST_DIR_CLEANUP` should be added to `passenv` for the same reason.